src/org/python/core/JavaProxyList.java

```diff
@@ -339,38 +371,17 @@ public PyObject __call__(PyObject key) {
     private static final PyBuiltinMethodNarrow listEqProxy = new ListMethod("__eq__", 1) {
         @Override
         public PyObject __call__(PyObject other) {
-            List jList = asList();
-            if (other.getType().isSubType(PyList.TYPE)) {
-                PyList oList = (PyList) other;
-                if (jList.size() != oList.size()) {
-                    return Py.False;
-                }
-                for (int i = 0; i < jList.size(); i++) {
-                    if (!Py.java2py(jList.get(i))._eq(oList.pyget(i)).__nonzero__()) {
-                        return Py.False;
-                    }
-                }
-                return Py.True;
-            } else {
-                Object oj = other.getJavaProxy();
-                if (oj instanceof List) {
-                    List oList = (List) oj;
-                    if (jList.size() != oList.size()) {
-                        return Py.False;
-                    }
-                    for (int i = 0; i < jList.size(); i++) {
-                        if (!Py.java2py(jList.get(i))._eq(
-                                Py.java2py(oList.get(i))).__nonzero__()) {
-                            return Py.False;
-                        }
-                    }
-                    return Py.True;
-                } else {
-                    return null;
-                }
-            }
+            return Py.newBoolean(isEqual(other));
         }
     };
+
+    private static final PyBuiltinMethodNarrow listNeProxy = new ListMethod("__ne__", 1) {
```

Comment:
Might the outer code `PyObject._ne, I assume, or is it `PyList._ne`, be able to do this. However, is there a logical error in that as postulated above.